### DISCUSSION TITLE
Any developer docs available?

### DISCUSSION THREAD

**Discussion by inix** [facebook/docusaurus#4300] [General] [answered]
Dear team:
I want to contribute to this repo, I also study codes of this repo,but so many modules that confuse me.
I am stucking at "How markdown files become result static files" now.
Do we have some resources, architecture design or some developer docs available?
Thanks.

**Comment by slorber** [comment, marked as answer]
Hi and thanks forr trying to contribute.

We don't have much architecture documentation for now, but it's something we have recently discussed and want to create, mostly because we often welcome interns, external contributors and MLH fellow interns, and explaining everything everytime is a time sink

Track  this here: https://github.com/facebook/docusaurus/issues/4301

---

About your question:

- plugins call read the FS and for each  md file they call:

```js
addRoute({
   path: "/docs/doc1",
   component: "@theme/DocPage"
   modules  {
     content: "./docs/doc1.mdx"
   }
})
```

- then the `@theme/DocPage` component will receive the doc as props:

```jsx
export default function DocPage({content}) {
  const DocContent  = content;
  return (
    <Layout>
      <DocContent>
    </Layout>
  );
}
```

- the mdx loader is responsible for compiling the doc at  `./docs/doc1.mdx` as a React component. The [MDX playground](https://mdxjs.com/playground) might help you understand this process.

- then Docusaurus core iterate over all JS pages and has a webpack configuration to emit the static HTML files for each page created by `addRoute()` calls

**Reply by inix** [reply]
Hi
Thanks for your detail reply and this excellent product.

**Reply by inix** [reply]
I have found where markdown files become JSX files.
Thanks you.

docusaurus/docusaurus-mdx-loader/src/index.js: 55
```javascript
  try {
    result = await mdx(content, options);
  } catch (err) {
    return callback(err);
  }
```
